### PR TITLE
[SPARK-42092][BUILD] Upgrade RoaringBitmap to 0.9.38

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims upgrade RoaringBitmap 0.9.38

### Why are the changes needed?
This version bring a bug fix:

- https://github.com/RoaringBitmap/RoaringBitmap/pull/613

a performance optimization:

- https://github.com/RoaringBitmap/RoaringBitmap/pull/612

other changes as follows:

https://github.com/RoaringBitmap/RoaringBitmap/compare/0.9.36...0.9.38

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### CODE REVIEW

File: core/benchmarks/MapStatusesConvertBenchmark-jdk17-results.txt [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 
-OpenJDK 64-Bit Server VM 17.0.5+8 on Linux 5.15.0-1023-azure
+OpenJDK 64-Bit Server VM 17.0.5+8 on Linux 5.15.0-1030-azure
 Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
 MapStatuses Convert:                      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Num Maps: 50000 Fetch partitions:500               1161           1178          18          0.0  1160681209.0       1.0X
-Num Maps: 50000 Fetch partitions:1000              2349           2394          47          0.0  2349393891.0       0.5X
-Num Maps: 50000 Fetch partitions:1500              3610           3651          39          0.0  3610385045.0       0.3X
+Num Maps: 50000 Fetch partitions:500               1116           1126           9          0.0  1115840641.0       1.0X
```
Comment (LuciferYang) [inline core/benchmarks/MapStatusesConvertBenchmark-jdk17-results.txt:9]: no significant performance difference between 0.9.36 and 0.9.38

**Review by yaooqinn — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by LuciferYang** [comment]
Thanks @srowen @dongjoon-hyun @yaooqinn

**Pull request closed without merging**